app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -80,11 +99,19 @@ def connect(options = {})
     username = options[:user] || authentication_userid(options[:auth_type])
     password = options[:pass] || authentication_password(options[:auth_type])
     service  = options[:service] || "Service"
-
-    result = self.class.raw_connect(server, port, path, username, password, service)
+    version  = options[:version] || 3
+
+    # Create the underlying connection according to the version of the oVirt API requested by
+    # the caller:
+    result =
```

Comment:
You could also do:

```ruby
connect_method = version == 4 ? :raw_connect_v4 : :raw_connect_v3
connection = class.public_send(connect_method, server, port, path, username, password, service)
default_endpoint.path = version == 4 ? '/ovirt-engine/api' : result.api_path
connection
```

But the way you have it, is also ok, besides `result` being too generic imho